extensions/amp-story/1.0/amp-story.js

```diff
@@ -723,6 +725,14 @@ export class AmpStory extends AMP.BaseElement {
       true /** callToInitialize */
     );
 
+    this.storeService_.subscribe(
```

Comment:
If `amp-story` runs after `amp-story-subscription` this piece of code will never execute, I recommend you pass the parameter `true` to `this.storeService.subscribe` (so the store calls this function immediately) and then check `if (pageIndex != -1) {resolve and set pageIndex}` in your handler

Reply:
Yes so if the `amp-story-subscriptions` runs first, no need to run this code at all, because we already get it in the store service.

That said, I will make the suggested changes since it might be able to delay the time we actually try to get the index from store service and might have a higher possibility that it is already resolved.

Reply:
Yeah, but if this happens between the constructor and the build callback it would fail, and also the deferred won't get resolved. Will resolve anyways bc it's fixed!